04-tidy.Rmd

```diff
@@ -177,7 +177,7 @@ Observe that `drinks_smaller` has three separate variables `beer`, `spirit`, and
 drinks_smaller_tidy
 ```
 
-Observe that while `drinks_smaller` and `drinks_smaller_tidy` are both rectangular in shape and contain the same 12 numerical values (3 alcohol types by 4 countries), they are formatted differently. `drinks_smaller` is formatted in what's known as \index{wide data format} ["wide"](https://en.wikipedia.org/wiki/Wide_and_narrow_data) format, whereas `drinks_smaller_tidy` is formatted in what's known as ["long/narrow"](https://en.wikipedia.org/wiki/Wide_and_narrow_data#Narrow) format.
+Observe that while `drinks_smaller` and `drinks_smaller_tidy` are both rectangular in shape and contain the same `r nrow(drinks_smaller_tidy)` numerical values (`r drinks_smaller_tidy %>% select(type) %>% n_distinct()` alcohol types by `r drinks_smaller_tidy %>% select(country) %>% n_distinct()` countries), they are formatted differently. `drinks_smaller` is formatted in what's known as \index{wide data format} ["wide"](https://en.wikipedia.org/wiki/Wide_and_narrow_data) format, whereas `drinks_smaller_tidy` is formatted in what's known as ["long/narrow"](https://en.wikipedia.org/wiki/Wide_and_narrow_data#Narrow) format.
```

Comment:
Since we use the values 12, 3, and 4 a few times, let's 
1. Run the computation to compute these values only once in an `echo=FALSE, purl=FALSE` block
1. Save these values in variables
1. Inline output the values using the variable names

Reply:
Taking this approach will also make raw bookdown Rmd code easier to read